picoquic/sockloop.c

```diff
@@ -440,8 +449,9 @@ int picoquic_packet_loop(picoquic_quic_t* quic,
             int sock_ret;
             int testing_nat = (ret == PICOQUIC_NO_ERROR_SIMULATE_NAT);
 
-            sock_ret = picoquic_packet_loop_open_sockets(0, sock_af[0], &s_mig, &s_mig_af,
-                &next_port, socket_buffer_size, 1);
+            sock_ret = picoquic_packet_loop_open_sockets(config->multipath_default_ip, 0,
+                                                         sock_af[0], &s_mig, &s_mig_af,
+                                                         &next_port, socket_buffer_size, 1);
             if (sock_ret != 1 || s_mig == INVALID_SOCKET) {
```

Comment:
See discussion. You don't need that. In fact, it is probably harmful. The system knows how to select the most appropriate IP for a destination. For example, if you call loopback, it will select loopback address as source. Binding sockets to fixed IP addresses will change that behavior.